### PR TITLE
refactor: Migrate participant-item to react

### PR DESCRIPTION
**Pull request by thisisamir98**

### CODE REVIEW

File: src/script/components/list/ParticipantItem.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+import {registerReactComponent} from 'Util/ComponentUtil';
+import ParticipantAvatar, {AVATAR_SIZE} from 'Components/ParticipantAvatar';
+import {UserlistMode} from 'Components/userList';
+import {t} from 'Util/LocalizerUtil';
+import {capitalizeFirstChar} from 'Util/StringUtil';
+
+import {User} from '../../entity/User';
+import {ServiceEntity} from '../../integration/ServiceEntity';
+// import {viewportObserver} from '../../ui/viewportObserver';
```
Comment (bennycode) [inline src/script/components/list/ParticipantItem.tsx]: leftover

File: src/script/components/list/ParticipantItem.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+      )}
+      <NamedIcon name="disclose-icon" className="disclose-icon" />
+      {/* END isInViewport() */}
+    </div>
+  );
+};
+
+export default ParticipantItem;
+
+registerReactComponent('participant-item', {
```
Comment (bennycode) [inline src/script/components/list/ParticipantItem.tsx]: Florian added generic types to `registerReactComponent`, so I guess you can do `registerReactComponent<ParticipantItemProps>` now?!

File: src/script/components/AvailabilityState.tsx [hunk 3]
```diff
@@ -119,6 +121,12 @@ const AvailabilityState: React.FC<AvailabilityStateProps> = ({
       )}
     </React.Fragment>
   );
+
+  if (className) {
```
Comment (bennycode) [inline src/script/components/AvailabilityState.tsx:125]: Do you need the `if (className)` when using `clsx`?

File: src/script/components/AvailabilityState.tsx [hunk 4]
```diff
@@ -27,6 +27,7 @@ import NamedIcon from './NamedIcon';
 
 export interface AvailabilityStateProps {
   availability: Availability.Type;
+  className: string;
```
Comment (bennycode) [inline src/script/components/AvailabilityState.tsx]: Is this optional? Because you have a `if (className)` check later on...

**Review by bennycode — CHANGES_REQUESTED** [review]

File: src/script/components/AvailabilityState.tsx (hunk 3 above)
Comment (thisisamir98) [reply src/script/components/AvailabilityState.tsx:125]: nope but this is solving some other issue, it adds a surrounding element based on this className

File: src/script/components/AvailabilityState.tsx (hunk 4 above)
Comment (thisisamir98) [reply src/script/components/AvailabilityState.tsx]: done in 
https://github.com/wireapp/wire-webapp/pull/10469/commits/e8aea880f2beff8c6d8c8426bf4a1ec6cf26f318

**Review by bennycode — APPROVED** [review]

File: src/script/components/list/ParticipantItem.tsx [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+          <NamedIcon name="disclose-icon" className="disclose-icon" />
+        </>
+      )}
+    </div>
+  );
+};
+
+export default ParticipantItem;
+
+registerReactComponent<ParticipantItemProps>('participant-item', {
```
Comment (ffflorian) [inline src/script/components/list/ParticipantItem.tsx:262]:
```suggestion
registerReactComponent('participant-item', {
```

**Review by AndyLnd — APPROVED** [review]

**Pull request merged**